src/main.rs

```diff
@@ -109,42 +104,52 @@ fn dropfile_filename(filename: &str) -> String {
 }
 
 fn write_new_door32sys_dropfile(
-    orig_contents: &str, out_path: &Path, socket_fd: i64) -> Result<String, String>
-{
+    orig_contents: &str,
+    out_path: &Path,
+    socket_fd: u64,
+) -> Result<String, Error> {
     if !out_path.is_dir() {
-        return Err(format!("{} is not a directory", out_path.to_string_lossy()));
+        return Err(Error::new(
+            ErrorKind::NotFound,
+            format!("{} is not a directory", out_path.to_string_lossy()),
+        ));
     }
 
-    let dropfile_path = out_path.join(dropfile_filename(DOOR32_SYS_FILENAME));
-
     //
     //  First two lines are as follows:
     //  1 - Comm type (2=telnet)
     //  2 - Shared socket fd
     //  ...the rest is just copied over from the original.
-    //      
-    let mut contents = format!("2\r\n{}\r\n", socket_fd);
-    let remaining_lines = orig_contents.lines().skip(2);
-    for line in remaining_lines {
-        contents.push_str(&format!("{}\r\n", line));
+    //
+    let dropfile_path = out_path.join(dropfile_filename(DOOR32_SYS_FILENAME));
+    let mut file = fs::File::create(&dropfile_path)?;
+    file.write_all(b"2\r\n")?;
+    file.write_all(socket_fd.to_string().as_bytes())?;
+    file.write_all(b"\r\n")?;
+    for line in orig_contents.lines().skip(2) {
+        let cp437 = &line.to_string().into_cp437(&CP437_CONTROL).map_err(|e| {
+            Error::new(
+                ErrorKind::InvalidData,
+                format!("Failed to convert to CP437: {}", e.into_string()),
+            )
+        })?;
+        file.write_all(cp437)?;
+        file.write_all(b"\r\n")?;
     }
-
-    match contents.to_string().into_cp437(&CP437_CONTROL) {
-        Ok(cp437)   => {
-            match fs::write(&dropfile_path, cp437) {
-                Ok(()) => {
-                    println!("Created new dropfile at {}", dropfile_path.to_string_lossy());
-                    Ok(dropfile_path.to_string_lossy().to_string())
-                },
-                Err(e) => Err(e.to_string()),
-            }
-        },
-        Err(e)      => Err(format!("Failed to convert {} to CP437", e.into_string())),
-    }    
+    let pathname = dropfile_path.to_string_lossy();
+    println!("Created new dropfile at {}", &pathname);
+    Ok(pathname.to_string())
 }
 
 #[cfg(windows)]
-fn get_socket_fd(stream: TcpStream) -> Result<i64, String> {
+fn get_socket_fd(stream: TcpStream) -> Result<u64, String> {
```

Comment:
I wonder about the `i64` vs `u64`: In many languages, it's become a best practice to avoid unsigned types due to their quirkyness. Is a file descriptor really going to need the entire unsigned space ever?

Reply:
It's not about range so much as trying to match the underlying types.  I looked at the expanded types for Windows's Handle and raw file descriptor's in the Unix case. The former expands to u64; the latter to c_int.

I arbitrarily chose `u64` to match Windows's representation. I don't really use windows, but I was under the impression that HANDLEs aren't necessarily integers as in Unix, but rather opaque values that might be pointers, in which case the high bit might be set. I could be wrong, though; `i64` would probably be fine here.